vulnerability/events_creator.go

```diff
@@ -45,7 +49,7 @@ type Vulnerability struct {
 	Description    string     `json:"description,omitempty"`
 	Severity       string     `json:"severity,omitempty"`
 	Classification string     `json:"classification,omitempty"`
-	PublishedDate  time.Time  `json:"published_date,omitempty"`
+	PublishedDate  *time.Time `json:"published_date,omitempty"`
```

Comment:
Bug fix - not all vulnerabilities contain `PublishedDate`, causing null pointer dereference.

Reply:
Maybe we shouldn't work with pointers directly in our structures and have something similar to this?
https://github.com/elastic/cloudbeat/pull/865

My initial solution was exactly like your fix here is the commit:
https://github.com/elastic/cloudbeat/pull/848/commits/e23845639bb60977afd37ee2968e0152908bde40

Reply:
@jeniawhite - that's a good solution as well.
IMHO, it will be a better approach to omit the field instead of sending the wrong data in case of a missing value.